From 5fb3493773b5c2f8b03f55ddf79520bb66cdbc45 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Fr=C3=A9d=C3=A9ric=20Pierret=20=28fepitre=29?=
 <frederic.pierret@qubes-os.org>
Date: Sun, 8 Nov 2020 16:17:29 +0100
Subject: [PATCH] Strip build path directories in tools, xen and xen/arch/x86

Ensure to have a realpath for XEN_ROOT else it fails to substitute
properly pathes in strings sections



-----------
diff --git a/tools/Rules.mk b/tools/Rules.mk
index ccde2d4..0982a67 100644
--- a/tools/Rules.mk
+++ b/tools/Rules.mk
@@ -140,6 +140,8 @@ endif
 CFLAGS-$(CONFIG_X86_32) += $(call cc-option,$(CC),-mno-tls-direct-seg-refs)
 CFLAGS += $(CFLAGS-y)
 
+CFLAGS += -ffile-prefix-map=$(XEN_ROOT)=.
+
 CFLAGS += $(EXTRA_CFLAGS_XEN_TOOLS)
 
 INSTALL_PYTHON_PROG = \
diff --git a/xen/Makefile b/xen/Makefile
index 5fba307..c96cb35 100644
--- a/xen/Makefile
+++ b/xen/Makefile
@@ -21,7 +21,7 @@ export PYTHON		?= $(PYTHON_INTERPRETER)
 export CHECKPOLICY	?= checkpolicy
 
 export BASEDIR := $(CURDIR)
-export XEN_ROOT := $(BASEDIR)/..
+export XEN_ROOT := $(realpath $(BASEDIR)/..)
 
 # Do not use make's built-in rules and variables
 MAKEFLAGS += -rR
@@ -259,6 +259,8 @@ ifneq ($(CONFIG_CC_IS_CLANG),y)
 CFLAGS += -Wa,--strip-local-absolute
 endif
 
+CFLAGS += -ffile-prefix-map=$(XEN_ROOT)=.
+
 AFLAGS += -D__ASSEMBLY__
 
 $(call cc-option-add,AFLAGS,CC,-Wa$(comma)--noexecstack)
diff --git a/xen/arch/x86/Makefile b/xen/arch/x86/Makefile
index 69b6cfa..f202984 100644
--- a/xen/arch/x86/Makefile
+++ b/xen/arch/x86/Makefile
@@ -125,6 +125,7 @@ ifneq ($(efi-y),)
 # Check if the compiler supports the MS ABI.
 export XEN_BUILD_EFI := $(shell $(CC) $(XEN_CFLAGS) -c efi/check.c -o efi/check.o 2>/dev/null && echo y)
 CFLAGS-$(XEN_BUILD_EFI) += -DXEN_BUILD_EFI
+CFLAGS-$(XEN_BUILD_EFI) += -ffile-prefix-map=$(XEN_ROOT)=.
 
 # Check if the linker supports PE.
 #EFI_LDFLAGS = $(patsubst -m%,-mi386pep,$(XEN_LDFLAGS)) --subsystem=10
-----------------
